### DISCUSSION TITLE
Used IRScrutinizer, but protocol not detected

### DISCUSSION THREAD

**Discussion by kEEs723** [bengtmartensson/IrScrutinizer#466] [Decoding issues] [answered]
Hi Bength Martensson, I am a happy user of you tool, the IR scrutinzer, v2.3.1. 
However, with this sequence, I have issues with decoding. 
I get this back, but I can't deduce which protocol I need to install to be able to decode it well. 

{38.0k,256,msb}<2,-3|2,-8>(9000u,-8,A:64,B:35){A=0x500518430305330,B=0x181818181}

+9000 -2700 +550 -800 +550 -750 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -800 +550 -2000 +500 -800 +550 -800 +550 -800 +550 -750 +600 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -800 +550 -1950 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -800 +550 -750 +550 -800 +550 -800 +550 -2000 +500 -800 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -2000 +550 -1950 +550 -800 +500 -850 +550 -800 +550 -800 +500 -800 +550 -2000 +550 -800 +500 -2000 +550 -800 +550 -800 +500 -2000 +550 -2000 +500 -800 +550 -800 +550 -2000 +500 -2000 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -2000 +550 -1950 +550 -800 +550 -800 +550 -800 +550 -800 +550 -750 +600 -750 +550 -2000 +550 -1950 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -1950
+250 -800 +550 -800 +550 -800 +500 -850 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -750 +600 -800 +500 -2000 +550 -800 +550 -1950 +550 -2000 +550 -750 +550 -2000 +550 -30100
+9000 -2650 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -2000 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -750 +600 -800 +550 -1950 +550 -800 +550 -1950 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -800 +500 -800 +550 -800 +550 -800 +550 -2000 +500 -800 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -2000 +550 -1950 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -2000 +550 -800 +500 -2000 +550 -800 +550 -750 +550 -2000 +550 -2000 +500 -800 +550 -800 +550 -2000 +550 -1950 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -750 +550 -2000 +550 -1950 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -2000 +500 -2000 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +500 -850 +500 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -2000
+350 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -1950 +550 -800 +550 -2000 +550 -1950 +550 -800 +550 -800 +550 -30100
+9000 -2650 +550 -800 +550 -800 +550 -800 +550 -750 +550 -800 +550 -2000 +500 -800 +550 -2000 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -750 +550 -800 +550 -800 +550 -1950 +550 -800 +550 -2000 +550 -800 +500 -800 +550 -800 +550 -2000 +500 -2000 +550 -800 +550 -800 +550 -750 +550 -800 +550 -2000 +550 -800 +500 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -1950 +550 -2000 +550 -750 +550 -800 +550 -800 +550 -800 +550 -800 +500 -2000 +550 -800 +550 -1950 +550 -800 +550 -800 +550 -1950 +550 -2000 +500 -850 +550 -800 +500 -2000 +550 -2000 +500 -800 +550 -800 +550 -800 +550 -800 +550 -750 +550 -800 +550 -2000 +550 -1950 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +500 -2000 +550 -2000 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -2000 +550 -2000 +550 -750 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -2000 +550 -1950 +550 -800 +550 -800 +500 -850 +550 -800 +500 -800 +550 -800 +550 -2000
+250 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -750 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +500 -850 +550 -800 +500 -800 +550 -800 +550 -800 +500 -850 +550 -750 +550 -800 +550 -800 +550 -800 +550 -800 +550 -750 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -800 +550 -800 +550 -800 +550 -800 +500 -800 +550 -2000 +550 -800 +550 -1950 +550 -2000 +550 -750 +550 -2000 +550 -30100

I have a decorative fireplace with a remote, of which I want to copy the commands to be able to controll it with an arduino.

**Comment by bengtmartensson** [comment]
Not all used IR signals correspond to previously known and documented protocols. It is quite possible, in particular for these kind of long signals, that it is something invented by the device manufacturer. Also, your captures are of fairly low quality (using a demodulating receiver probably). Consider getting better capturing hardware, using a non-demodulating sensor, like for example [this](http://www.harctoolbox.org/arduino_nano.html). Then capture all the available commands, collect the results in a text file containing the command names and the captures. Try to find out if the keys are repeating or not. Then post the result here, and we'll see we what we can do.

**Comment by kEEs723** [comment]
Thanks for answering! 
From the past I have experience with IR, I even built small solutions like operating room lights with an IR transmitter. With the right black transmitter, I get very consistent outputs. (I have also checked its output with your scrutinizer, it was NEC: {D=0,F=70}, beg=0, end=67) 
![image](https://user-images.githubusercontent.com/5416007/208997604-13f2aeb9-b319-46ae-a3c0-5ff1a75ab25d.png)
My solution is what I always used, namely an arduino nano with a standard [receiver](https://www.conrad.nl/nl/p/iduino-1485322-infrarood-ontvanger-geschikt-voor-arduino-1-stuk-s-1485322.html?utm_medium=affiliate&utm_source=tradetracker&utm_campaign=316050&utm_content=ShopForward+NL+Shopping)
![image](https://user-images.githubusercontent.com/5416007/208997935-1ece6d5c-73e0-4295-a59c-08c4250ef975.png)
You asked for more output. I have recorded the outputs for every button in a txt file. 
To be 100% clear, I gave each button a number: 
![image](https://user-images.githubusercontent.com/5416007/208998690-8ecba003-9126-431f-b9db-3f7ec03c4da7.png)
I have added the code that is outputted when 7 buttons are pressed. The rest of them is just for settings on the remote itself. 
[ButtonIRidentifications.txt](https://github.com/bengtmartensson/IrScrutinizer/files/10281428/ButtonIRidentifications.txt)

**Comment by bengtmartensson** [comment, marked as answer]
I edited the file to a format that we can analyze, found [here](https://gist.github.com/bengtmartensson/4697477b577c62e58413ef35f1121cb9). Running irptransmogrifier/analyze  on that file gives:

```
$ irptransmogrifier  analyze --named ButtonIRidentifications.txt -M 256 --timebase 562
On$1 {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x21005a2e18181818181818181,B=0x2a}
On$2 {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x21805a2e18181818181818181,B=0x2b}
On$3 {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x21005a2e18181818181818181,B=0x2a}
On$4 {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x21805a2e18181818181818181,B=0x2b}
On$5 {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x21005a2e18181818181818181,B=0x2a}
On$6 {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x21805a2e18181818181818181,B=0x2b}
1$1  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x25803a2018181818181818181,B=0x200000000b7}
1$2  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x25803a2018181818181818181,B=0x200000000b7}
1$3  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x25803a2018181818181818181,B=0x200000000b7}
1$4  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x25803a2018181818181818181,B=0x200000000b7}
1$5  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x25803a2018181818181818181,B=0x200000000b7}
2$1  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x21803a60181818181818181819000000006c}
2$2  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x26803a601818181818181818190000000064}
2$3  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x22803a601818181818181818190000000068}
2$4  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x24803a601818181818181818190000000060}
2$5  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x24803a601818181818181818190000000060}
3$1  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x44c03a3018181818181818181,B=0x3000000008c}
3$2  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x44a03a3018181818181818181,B=0x3000000004c}
3$3  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x44e03a3018181818181818181,B=0x300000000cc}
3$4  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x44903a7018181818181818181,B=0x300000000ac}
3$5  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,206u,-802u,B:43,1,-30m){A=0x44d03a7018181818181818181,B=0x3000000006c}
4$1  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x4803a1018181818181818181400000000b1}
4$2  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m){A=0x44803a1018181818181818181,B=0x40000000071}
4$3  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x24803a1018181818181818181400000000f1}
4$4  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x4803a1018181818181818181400000000b1}
4$5  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x44803a101818181818181818140000000071}
5$1  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x44903a7018181818181818181d00000000b4}
5$2  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x44e03a7018181818181818181d0000000034}
5$3  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x44a03a7018181818181818181d00000000d4}
5$4  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x44c03a7018181818181818181d0000000054}
5$5  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m){A=0x44803a7018181818181818181d0000000094}
6$1  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-2009u,B:43,1,-30m){A=0x44803a5018181818181818181,B=0x100000000ff}
6$2  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-2009u,B:43,1,-30m){A=0x44803a5018181818181818181,B=0x100000000ff}
6$3  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-2009u,B:43,1,-30m){A=0x44803a5018181818181818181,B=0x100000000ff}
6$4  {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-2009u,B:43,1,-30m){A=0x44803a5018181818181818181,B=0x100000000ff}
```
So there are a few questions left. In particular, why are some signals a payload of 143 bits, while others are "broken up" after 99 bits? The former can be used as a protocol {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:143,1,-30m), the rest with a similar protocol {562,msb}<1,-802u|1,-2009u>(16,-2009u,A:99,354u,-802u,B:43,1,-30m). But I have a bad feeling that your captures are too imprecise.  Using a demodulating receiver like yours is not a good idea for analyzing IR signals, if possible get better equipment, like [this](www.harctoolbox.org/arduino_nano.html). (I do not hold very much of "hobbyist IR boards" like the one you linked. It is in general a low-quality Chinese TR sensor (for example VS1838), with some non-essential resistors etc. Better to by for example a good-quality sensor from Vishay, they cost less than your hobbyist board, and comes with a data sheet.

You can now go on trying to identify which bit makes what in the IRP representation. You will probably find IrpTransmogrifier usefule for this, a part of IrScrutinizer, and installed (as wrapper) together with the latter.

**Reply by kEEs723** [reply]
Thanks for your post and answers! I have ordered the right hardware (indicated in your links) now to see if the consistency improves. I expect results in 1-2 weeks.

**Comment by kEEs723** [comment]
Hi, I was able to use the appropiate hardware, namely the TSMP58000. Just connected the bare minimum and I was able to use the .bat file to flash GirsLite on my nano (following this guide: http://www.harctoolbox.org/arduino_nano.html) 
![image](https://user-images.githubusercontent.com/5416007/212769014-7cac7b8b-8d20-4afe-9944-b719fc476d2e.png)
I used IR scrutinizer at "capturing hw" and captured 5x the data for each button. Now there seems to come some consistency in the data!, however, no decodes. 
[ButtonIRidentificationsGoodHW.txt](https://github.com/bengtmartensson/IrScrutinizer/files/10429626/ButtonIRidentificationsGoodHW.txt)

**Comment by bengtmartensson** [comment]
Now the captures all look much more similar; all signals seem to decode to `{35,msb}<16,-792u|16,-2005u>(9109u,-2005u,A:192,16,-65m)`.  (So the purcase of the TSMP58000 was worth it!) So if you add this to the protocol data base IrpProtocols.xml, you will see that they all decode to that protocol. 

I am not going to clean up your log file for you. If you want to pursue this issue further, you can capture the signals in raw mode in IrScrutinizer, using the Scrutinizer remote -> Raw remote pane. If you so desire, you can try to identify what the different of the 192 bits do. Strange is that subsequent captures of the "same" command seem to differ in as much as 9 bits: Sometimes there are toggles, they are in general one bit. Possibly there are time stamps, or battery status, air pressure/temperature/humidity or anything ... 

But this is probably beyond the original question.
